### ISSUE TITLE
Changing calculated column type: 'Update formula' menu entry is missing

### ISSUE THREAD

**Issue by katerina-stepanova** [datagrok-ai/public#1558]
DG 1.12.2

**Steps to reproduce:**
1. Create a dataframe and add a calculated column:
```
let df = DG.DataFrame.fromColumns([
 DG.Column.fromList(DG.TYPE.FLOAT, 'x', [1, 2, 3]),
 DG.Column.fromList(DG.TYPE.FLOAT, 'y', [4, 5, 6]),
 DG.Column.fromList(DG.TYPE.FLOAT, 'z', [7, 8, 9])
]);
let col = await df.columns.addNewCalculated('new', '${x}+${y}-${z}');
grok.shell.addTableView(df);
```
2. Open edit formula dialog:
`col.dialogs.editFormula();`
3. Change column type and click OK
4. Try to open edit formula dialog for this column again:
`col.dialogs.editFormula();`

**Expected results:** Edit formula dialog is opened again. Column can be edited and is recognised as a calculated column.
**Actual results:**
1. Dialog is not opened. 
2. Something's wrong with events used on our side to track calculated columns changes, calculated columns can get lost (TBD - more details from Andreas)

**Comment by alex-aprm** [comment]
It works right now.

Please, note, if column type changes, DG replaces original column, and "col" variable becomes obsolete. You need to find a new column by name.

**Comment by onuf** [comment]
@alex-aprm please confirm: is this included in `v1.13`?
